submissions/ses3332002/tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,97 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Being {
+  constructor (name, friends, species) {
+    this.species = species;
+    this.name = name;
+    this.friends = friends;
+  }
+  
+  getDescription() {
+    return Object.entries(this)
+    .filter(([key]) => propsToPrint.includes(key))
+    .map(([,value]) => Array.isArray(value) ? value.join(', ') : value)
+    .join('; ');
+  }
+};
+
+class Animal extends Being {
+  constructor (name, friends, gender, saying, species) {
+    super(name, friends, species);
+    this.gender = gender;
+    this.saying = saying;
+    this.legs = 4;
+    this.hands = 0;
+  }
+};
+
+class Dog extends Animal {
+  constructor (name, friends, gender) {
+    super(name, friends, gender);
+    this.species = "dog";
+    this.saying = "Woof-woof!";
+  }
+};
+
+class Cat extends Animal {
+  constructor (name, friends, gender) {
+    super(name, friends, gender);
+    this.species = "cat";
+    this.saying = "Meow!";
+  }
+};
+
+class Human extends Animal {
+  constructor (name, friends, gender, saying) {
+    super(name, friends, gender, saying);
+    this.species = "human";
+    this.legs = 2;
+    this.hands = 2;
+  }
+};
+
+class Superhero extends Human {
+  constructor (name, friends, gender, saying) {
+    super(name, friends, gender);
+    this.species = "super-hero";
+  }
+};
+
+const dog = new Dog ("Toby", ["Puss"], "male");
+const cat = new Cat ("Puss", ["Toby", "Halley"], "male");
+const man = new Human ("Jack", ["Jennie", "Halley"], "male", "Hey baby!");
+const woman = new Human ("Jennie", ["Jack", "Toby"], "female", "Hi darling!");
+const catWoman = new Superhero ("Halley", ["Puss"], "female");
```

Comment:
`Superhero` constructor expects `saying`. Never sent to it, never used inside it.

Reply:
Different instances of Superhero class has different saying properties: Superhero catWoman says "Meow!" exactly like a cat, but Superhero superMan says... something else. I can send saying property to constructor, but can i send "new Cat().saying" to it, like "const catWoman = new Superhero ("Halley", ["Puss"], "female", new Cat().saying);"?

Reply:
> I can send saying property to constructor, but can i send "new Cat().saying" to it, like "const catWoman = new Superhero ("Halley", ["Puss"], "female", new Cat().saying);"?

Did you try?

The problem is hinted in the second part of the statement: Never sent to it, **never used inside it**.

Reply:
I remade it.
Now all props are given via constructor.